libsigrokdecode fails to find Nix python3, falls back to non-nix python

**Describe the bug**
When running Pulseview from Nix, it is missing some decoders, notably uart. Looking a the stdout output, the following printed for a bunch of decoders
```
srd: ModuleNotFoundError: Failed to load decoder <decoder>: import by name failed: No module named 'math'
``` 
which is weird, since the "math" module is part of the Python stdlib. With strace, I found the following:
```
pulseview.log.990731:openat(AT_FDCWD, "/nix/store/6cfajs6lsy9b4wxp3jvyyl1g5x2pjmpr-python3-3.8.9/lib/libpython3.8.so.1.0", O_RDONLY|O_CLOEXEC) = 6
pulseview.log.990731:stat("/nix/store/xmxgxig6zxrixicc7905ssgb4yc3lysa-bash-interactive-4.4-p23/bin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
<SNIP: here some more /nix/ paths are tried for /bin/python3>
pulseview.log.990731:stat("/home/user/.nix-profile/bin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
pulseview.log.990731:stat("/nix/var/nix/profiles/default/bin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
pulseview.log.990731:stat("/usr/local/sbin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
pulseview.log.990731:stat("/usr/local/bin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
pulseview.log.990731:stat("/usr/sbin/python3", 0x7ffd4ffcdb10) = -1 ENOENT (No such file or directory)
pulseview.log.990731:stat("/usr/bin/python3", {st_mode=S_IFREG|0755, st_size=5490352, ...}) = 0
```
It tries to find the first available python executable, and since I don't have it in my .nix-profile, it falls back to my non-Nix system `python3`. Then all the imports fail. 
As a workaround, installing `python3` with nix-env resolves the problem 

**To Reproduce**
Steps to reproduce the behavior:
1. Remove python from nix-env: `nix-env -e python3`
2. Run pulseview: `nix-shell -p pulseview --run pulseview`
3. Observe import failures and missing decoders

**Expected behavior**
No import failures 

**Notify maintainers**
@bjornfor 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.0-77-generic, Ubuntu, 20.04.2 LTS (Focal Fossa)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.14`
 - channels(user): `"home-manager, nixpkgs-21.11pre300649.3e0ce8c5d47"`
 - channels(root): `"nixpkgs-21.11pre301805.dac74fead87"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

@bjornfor there's updates for some of the packages you maintain.
here's something to get you started https://github.com/Artturin/nixpkgs/commit/06caa4d5cfd3d76be127538093a3e5abcdc84064

building pulseview fails with
```
                 from /nix/store/dnld9lcxf0spng9kvgyc7yg15dmpd2v4-qtbase-5.14.2-dev/include/QtWidgets/QMenu:1,
                 from /build/pulseview-0.4.2/pv/widgets/decodermenu.hpp:23,
                 from /build/pulseview-0.4.2/pv/widgets/decodermenu.cpp:24:
/nix/store/dnld9lcxf0spng9kvgyc7yg15dmpd2v4-qtbase-5.14.2-dev/include/QtCore/qvariant.h:528:17: note: declared here
  528 | inline QVariant qVariantFromValue(const T &t)
      |                 ^~~~~~~~~~~~~~~~~
[ 95%] Building CXX object CMakeFiles/pulseview.dir/qrc_pulseview.cpp.o
[ 96%] Building CXX object CMakeFiles/pulseview.dir/qrc_translations.cpp.o
[ 97%] Building CXX object CMakeFiles/pulseview.dir/pv/views/viewbase.cpp.o
[ 98%] Building CXX object CMakeFiles/pulseview.dir/pv/views/trace/cursor.cpp.o
[100%] Linking CXX executable pulseview
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `_Py_Dealloc'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyLong_FromLong'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyModule_Create2'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_SetAttrString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyType_IsSubtype'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_Fetch'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyList_Insert'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyType_GetFlags'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyEval_RestoreThread'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_Size'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyUnicode_FromString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_HasAttrString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyBytes_AsString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyList_SetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_Clear'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyArg_ParseTupleAndKeywords'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyCallable_Check'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyBytes_Size'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySequence_Size'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyExc_TypeError'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyEval_InitThreads'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyFloat_FromDouble'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyImport_GetModuleDict'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySet_New'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyTuple_GetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyLong_AsLongLong'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyTuple_SetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyList_Size'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_GetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyBytes_AsStringAndSize'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyTuple_New'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyImport_AppendInittab'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_CallFunctionObjArgs'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_Occurred'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyUnicode_AsUTF8String'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `_Py_FalseStruct'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_SetItemString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyLong_Type'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_CallFunction'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyGILState_Ensure'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_SetString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_CallObject'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyList_GetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySys_GetObject'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyUnicode_FromStringAndSize'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyArg_ParseTuple'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyLong_AsSize_t'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_NormalizeException'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_Next'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_IsSubclass'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `_Py_NoneStruct'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_New'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyImport_Import'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `Py_Finalize'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `Py_DecRef'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `Py_InitializeEx'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyGILState_Release'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySet_Add'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyModule_AddObject'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySequence_GetItem'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyType_GenericNew'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `Py_BuildValue'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyDict_GetItemString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyBool_FromLong'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyLong_AsLong'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyExc_Exception'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `_Py_TrueStruct'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyFloat_Type'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyExc_IndexError'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyFloat_AsDouble'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySet_Pop'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyType_FromSpec'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyEval_SaveThread'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PySequence_Check'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `Py_IsInitialized'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_CallMethod'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_GetAttrString'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyModule_AddIntConstant'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyList_New'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyObject_Str'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyErr_Format'
/nix/store/fh9f7pr9kxfwvs8lm4479lvhrkzmmkdw-binutils-2.35.1/bin/ld: /nix/store/znpsf6hdmm03s9ps7zjx664kh2yq25w2-libsigrokdecode-0.5.3/lib/libsigrokdecode.so: undefined reference to `PyTuple_Size'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/pulseview.dir/build.make:1689: pulseview] Error 1
make[1]: *** [CMakeFiles/Makefile2:118: CMakeFiles/pulseview.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
builder for '/nix/store/dpq1c12vyxqhhq14fcdi62d6k4d6fgfc-pulseview-0.4.2.drv' failed with exit code 2
```

--- a/pkgs/development/tools/libsigrokdecode/default.nix
+++ b/pkgs/development/tools/libsigrokdecode/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, pkg-config, glib, python3, libsigrok, check }:
+{ lib, stdenv, fetchurl, pkg-config, glib, python3, check }:
 
 stdenv.mkDerivation rec {
   pname = "libsigrokdecode";
@@ -10,7 +10,9 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ glib python3 libsigrok check ];
+  buildInputs = [ glib python3 ];
+  checkInputs = [ check ];
+  doCheck = true;
 
   meta = with lib; {
     description = "Protocol decoding library for the sigrok signal analysis software suite";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -13455,7 +13455,9 @@ in
     sha256 = "0l3h7zvn3w4c1b9dgvl3hirc4aj1csfkgbk87jkpl7bgl03nk4j3";
   };
 
-  libsigrokdecode = callPackage ../development/tools/libsigrokdecode { };
+  libsigrokdecode = callPackage ../development/tools/libsigrokdecode {
+    python3 = python38;
+  };
 
   # special forks used for dsview
   libsigrok4dsl = callPackage ../applications/science/electronics/dsview/libsigrok4dsl.nix { };

